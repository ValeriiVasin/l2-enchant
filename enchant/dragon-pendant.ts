import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

export const dragonPendantEnchantMap = new Map([
  ...createEnchantMap({
    strategy: EnchantStrategy.Drop,
    enchantItem: Item.DragonVarnish,
    chances: [100, 90, 80, 70, 70],
    items: [
      Item.DragonPendantOne,
      Item.DragonPendantOne_1,
      Item.DragonPendantOne_2,
      Item.DragonPendantOne_3,
      Item.DragonPendantOne_4,
      Item.DragonPendantOne_5,
    ],
  }),

  [
    Item.DragonPendantOne_5,
    {
      item: Item.DragonPendantOne_5,
      cost: new Map([[Item.Adena, 12_000_000]]),
      results: [{ chance: 100, item: Item.DragonPendantTwo }],
    },
  ],

  ...createEnchantMap({
    strategy: EnchantStrategy.Drop,
    enchantItem: Item.DragonVarnish,
    chances: [100, 80, 70, 60, 30],
    items: [
      Item.DragonPendantTwo,
      Item.DragonPendantTwo_1,
      Item.DragonPendantTwo_2,
      Item.DragonPendantTwo_3,
      Item.DragonPendantTwo_4,
      Item.DragonPendantTwo_5,
    ],
  }),

  [
    Item.DragonPendantTwo_5,
    {
      item: Item.DragonPendantTwo_5,
      cost: new Map([
        [Item.Adena, 16_000_000],
        [Item.DragonVarnish, 3],
      ]),
      results: [{ chance: 100, item: Item.DragonPendantThree }],
    },
  ],

  ...createEnchantMap({
    strategy: EnchantStrategy.Drop,
    enchantItem: Item.DragonVarnish,
    chances: [70, 50, 30, 12, 5],
    items: [
      Item.DragonPendantThree,
      Item.DragonPendantThree_1,
      Item.DragonPendantThree_2,
      Item.DragonPendantThree_3,
      Item.DragonPendantThree_4,
      Item.DragonPendantThree_5,
    ],
  }),
]);
