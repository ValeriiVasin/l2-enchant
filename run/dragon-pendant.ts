import { enchant } from '../helpers/enchant';
import { header } from '../helpers/header';
import { Item } from '../items';

(function one() {
  const enchants: Array<[number, Item]> = [
    [1, Item.DragonPendantOne_1],
    [2, Item.DragonPendantOne_2],
    [3, Item.DragonPendantOne_3],
    [4, Item.DragonPendantOne_4],
    [5, Item.DragonPendantOne_5],
  ];

  console.log();
  console.log(header('Подвеска Дракона Ур.1'));
  for (const [lvl, item] of enchants) {
    console.log(
      `Подвеска Дракона Ур.1 +0 => +${lvl}`,
      enchant(Item.DragonPendantOne, item, 1000),
    );
  }
})();

(function two() {
  const enchants: Array<[number, Item]> = [
    [1, Item.DragonPendantTwo_1],
    [2, Item.DragonPendantTwo_2],
    [3, Item.DragonPendantTwo_3],
    [4, Item.DragonPendantTwo_4],
    [5, Item.DragonPendantTwo_5],
  ];

  console.log();
  console.log(header('Подвеска Дракона Ур.2'));
  for (const [lvl, item] of enchants) {
    console.log(
      `Подвеска Дракона Ур.2 +0 => +${lvl}`,
      enchant(Item.DragonPendantTwo, item, 1000),
    );
  }
})();

(function three() {
  const enchants: Array<[number, Item]> = [
    [1, Item.DragonPendantThree_1],
    [2, Item.DragonPendantThree_2],
    [3, Item.DragonPendantThree_3],
    [4, Item.DragonPendantThree_4],
    [5, Item.DragonPendantThree_5],
  ];

  console.log();
  console.log(header('Подвеска Дракона Ур.3'));
  for (const [lvl, item] of enchants) {
    console.log(
      `Подвеска Дракона Ур.3 +0 => +${lvl}`,
      enchant(Item.DragonPendantThree, item, 1000),
    );
  }
})();
