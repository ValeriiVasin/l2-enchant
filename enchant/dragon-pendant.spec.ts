import { dragonPendantEnchantMap } from './dragon-pendant';

it('enchant map: dragon pendant', () => {
  expect(dragonPendantEnchantMap).toMatchSnapshot();
});
