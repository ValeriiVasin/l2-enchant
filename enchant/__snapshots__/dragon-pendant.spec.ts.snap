// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: dragon pendant 1`] = `
Map {
  "Подвеска Дракона Ур.1" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.1",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Подвеска Дракона Ур.1 +1",
      },
      Object {
        "chance": 0,
        "item": "Подвеска Дракона Ур.1",
      },
    ],
  },
  "Подвеска Дракона Ур.1 +1" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.1 +1",
    "results": Array [
      Object {
        "chance": 90,
        "item": "Подвеска Дракона Ур.1 +2",
      },
      Object {
        "chance": 10,
        "item": "Подвеска Дракона Ур.1",
      },
    ],
  },
  "Подвеска Дракона Ур.1 +2" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.1 +2",
    "results": Array [
      Object {
        "chance": 80,
        "item": "Подвеска Дракона Ур.1 +3",
      },
      Object {
        "chance": 20,
        "item": "Подвеска Дракона Ур.1",
      },
    ],
  },
  "Подвеска Дракона Ур.1 +3" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.1 +3",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Подвеска Дракона Ур.1 +4",
      },
      Object {
        "chance": 30,
        "item": "Подвеска Дракона Ур.1",
      },
    ],
  },
  "Подвеска Дракона Ур.1 +4" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.1 +4",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Подвеска Дракона Ур.1 +5",
      },
      Object {
        "chance": 30,
        "item": "Подвеска Дракона Ур.1",
      },
    ],
  },
  "Подвеска Дракона Ур.1 +5" => Object {
    "cost": Map {
      "Адена" => 12000000,
    },
    "item": "Подвеска Дракона Ур.1 +5",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Подвеска Дракона Ур.2",
      },
    ],
  },
  "Подвеска Дракона Ур.2" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.2",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Подвеска Дракона Ур.2 +1",
      },
      Object {
        "chance": 0,
        "item": "Подвеска Дракона Ур.2",
      },
    ],
  },
  "Подвеска Дракона Ур.2 +1" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.2 +1",
    "results": Array [
      Object {
        "chance": 80,
        "item": "Подвеска Дракона Ур.2 +2",
      },
      Object {
        "chance": 20,
        "item": "Подвеска Дракона Ур.2",
      },
    ],
  },
  "Подвеска Дракона Ур.2 +2" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.2 +2",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Подвеска Дракона Ур.2 +3",
      },
      Object {
        "chance": 30,
        "item": "Подвеска Дракона Ур.2",
      },
    ],
  },
  "Подвеска Дракона Ур.2 +3" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.2 +3",
    "results": Array [
      Object {
        "chance": 60,
        "item": "Подвеска Дракона Ур.2 +4",
      },
      Object {
        "chance": 40,
        "item": "Подвеска Дракона Ур.2",
      },
    ],
  },
  "Подвеска Дракона Ур.2 +4" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.2 +4",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Подвеска Дракона Ур.2 +5",
      },
      Object {
        "chance": 70,
        "item": "Подвеска Дракона Ур.2",
      },
    ],
  },
  "Подвеска Дракона Ур.2 +5" => Object {
    "cost": Map {
      "Адена" => 16000000,
      "Лак Дракона" => 3,
    },
    "item": "Подвеска Дракона Ур.2 +5",
    "results": Array [
      Object {
        "chance": 100,
        "item": "Подвеска Дракона Ур.3",
      },
    ],
  },
  "Подвеска Дракона Ур.3" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.3",
    "results": Array [
      Object {
        "chance": 70,
        "item": "Подвеска Дракона Ур.3 +1",
      },
      Object {
        "chance": 30,
        "item": "Подвеска Дракона Ур.3",
      },
    ],
  },
  "Подвеска Дракона Ур.3 +1" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.3 +1",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Подвеска Дракона Ур.3 +2",
      },
      Object {
        "chance": 50,
        "item": "Подвеска Дракона Ур.3",
      },
    ],
  },
  "Подвеска Дракона Ур.3 +2" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.3 +2",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Подвеска Дракона Ур.3 +3",
      },
      Object {
        "chance": 70,
        "item": "Подвеска Дракона Ур.3",
      },
    ],
  },
  "Подвеска Дракона Ур.3 +3" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.3 +3",
    "results": Array [
      Object {
        "chance": 12,
        "item": "Подвеска Дракона Ур.3 +4",
      },
      Object {
        "chance": 88,
        "item": "Подвеска Дракона Ур.3",
      },
    ],
  },
  "Подвеска Дракона Ур.3 +4" => Object {
    "cost": "Лак Дракона",
    "item": "Подвеска Дракона Ур.3 +4",
    "results": Array [
      Object {
        "chance": 5,
        "item": "Подвеска Дракона Ур.3 +5",
      },
      Object {
        "chance": 95,
        "item": "Подвеска Дракона Ур.3",
      },
    ],
  },
}
`;
